docs/DeveloperGuide.md

```diff
@@ -2,37 +2,189 @@
+  1. Retrieves and validates the matriculation number and component description from `arguments`.
+  2. Confirms the existence of the specified component and student.
+  3. Locates and deletes the `Grade` object from the `GradeList` in `AppState`.
+
+#### Example Usage Scenario
+
+Given below is an example usage scenario and how the AddGradeCommand and DeleteGradeCommand behaves at each step.
+
+Consider a scenario where the Professor wants to add a grade for a student, Alice, in the midterm component of a module. The Professor would initiate the AddGradeCommand, inputting Alice's matriculation number, the component as "midterm," and a score of 85. The command will then:
+
+1. Validate Alice's matriculation number and ensure that the midterm component is present in the grade configuration for the module.
+2. Confirm that the score of 85 is within the allowable range for midterm.
+3. Create a new Grade object and add it to the GradeList in AppState, reflecting Alice's score for the midterm component.
+
+The sequence diagram of the DeleteGradeCommand is shown below.
+
+![AddGradeCommand.png](diagrams%2FAddGradeCommand.png)
+
+Later, if the Professor realizes that the grade was entered incorrectly and decides to delete it, they can use the DeleteGradeCommand:
+
+1. The command verifies Alice's matriculation number and the midterm component's presence.
+2. It locates Alice's midterm Grade object and deletes it from the GradeList in AppState, effectively removing the grade record.
+
+The sequence diagram of the DeleteGradeCommand is shown below.
+
+![DeleteGradeCommand.png](diagrams%2FDeleteGradeCommand.png)
+
+### Storage Load feature
+
+The `StudentStorage`, `GradeStorage` and `ComponentStorage` classes implement the feature to load data from the 
+data `.txt` files into their respective List objects at the start of the program.
+
+The load list methods for the Storage classes have largely similar logic flows.  
+The following section and sequence diagram elaborate on the implementation of the `loadGradeList` method in `GradeStorage`:
+
+![GradeStorage.png](diagrams/GradeStorage.png)
+
+1. TutorLink constructs a new `GradeStorage`.
+2. TutorLink calls `loadGradeList`.
+3. `GradeStorage` creates a new `ArrayList` of `Grade`s.
+4. `GradeStorage` creates a new `Scanner` with the path to the file.
+5. While there are next lines in the data file:
+   - `Scanner` returns the current file line as a String and moves to the next file line.
+   - `GradeStorage` calls its `getGradeFromFileLine` method with the file line.
+   - If the file line references a valid `Component` and a valid `Student`, a `Grade` is returned and added to the `ArrayList`.
+   - If not (e.g. if data file was corrupted), the file line is simply ignored, and the loop continues to the next iteration.
+6. The `ArrayList` of `Grade`s is returned to TutorLink.
```

Comment:
should this be more concise instead of going through your entire program/algorithm